DataEnrich/main.bal

```diff
@@ -0,0 +1,48 @@
+import ballerina/http;
+import ballerinax/salesforce;
+
+configurable string clientId = ?;
+configurable string clientSecret = ?;
+configurable string refreshToken = ?;
+configurable string refreshUrl = ?;
+configurable string baseUrl = ?;
+
+salesforce:ConnectionConfig sfConfig = {
+    baseUrl,
+    auth: {
+        clientId,
+        clientSecret,
+        refreshToken,
+        refreshUrl
+    }
+};
+
+type LeadInfoRequst record {|
+    string id;
+    string name;
+|};
+
+type EnrichedLead record {|
+    string id;
+    string name;
+    string email;
+    string company;
+    string status;
+    string phone;
+|};
+
+service /sales on new http:Listener(8080) {
+
+    resource function post leads/[string id](LeadInfoRequst req) returns EnrichedLead|error {
+        salesforce:Client baseClient = check new (sfConfig);
+        record {} res = check baseClient->getById("Lead", id, ["Id", "Name", "Email", "Company", "Status", "Phone"]);
```

Comment:
instead of open and empty record shall we use a record with `ID,Name,Email,Company,Status,Phone` as fields